test/TritonGPU/loop-pipeline.mlir

```diff
@@ -283,16 +283,10 @@ tt.func @lut_bmm_scalar(%77: i64 {tt.divisibility=16: i32},
 // CHECK: triton_gpu.insert_slice_async
 // CHECK: triton_gpu.insert_slice_async
 // CHECK: triton_gpu.insert_slice_async
-// CHECK: triton_gpu.async_commit_group
-// CHECK: %[[NEXT_BUFFER_1:.*]] = tt.addptr %{{.*}}, {{.*}}
-// CHECK: triton_gpu.insert_slice_async %[[NEXT_BUFFER_1]]
-// CHECK: %[[LUT_BUFFER_0:.*]] = tt.load %{{.*}}, {{.*}}
-// CHECK: %[[LUT_BUFFER_1:.*]] = tt.expand_dims %[[LUT_BUFFER_0]] {axis = 1 : i32}
-// CHECK: %[[LUT_BUFFER_2:.*]] = tt.broadcast %[[LUT_BUFFER_1]]
-// CHECK: %[[LUT_BUFFER_3:.*]] = arith.muli {{.*}}, %[[LUT_BUFFER_2]]
-// CHECK: %[[NEXT_BUFFER_0:.*]] = tt.addptr {{.*}}, %[[LUT_BUFFER_3]]
-// CHECK: triton_gpu.insert_slice_async %[[NEXT_BUFFER_0]]
-// CHECK: triton_gpu.async_wait {num = 2 : i32}
+// CHECK: triton_gpu.insert_slice_async
+// CHECK: triton_gpu.insert_slice_async
+// CHECK: triton_gpu.insert_slice_async
+// CHECK: triton_gpu.async_wait {num = 3 : i32}
```

Comment:
This is because the pipelining of a load that is not directly connected to the dot operation.

Reply:
Oh, this would have worse perf probably than no pipelining. You will be issuing async load for the lut load and then blocking right after until it's done, then you would have to do smem to distributed conversion. You will only get benefit if the async copies of dependent loads are > 1 iteration away from each other. @pawelszczerbuk is working on a PR for this.

Reply:
Also, the `lit` checks are not too informative, I would generally advise to keep some sort of def/use in the checks to make it clear what we are checking for (i.e. similar to the checks that you deleted).

Reply:
I see. Sounds like I should disable pipelining such loads that are dependencies of other mma loads. However for non-dependency load, such as `t4` below, pipelining it can be beneficial?
```

t1 = tt.load(A)
t2 = tt.load(B)
t3 = tt.dot(t1,t2)
t4 = tt.load(C)
t5 = t3 + t5

```

Reply:
>  should disable pipelining such loads that are dependencies of other mma loads

I guess the point is any chained loads would probably have worse perf if we do not space those async copies > 1 iterations apart, regardless if the last load is mma load or not. And yea I think the `t4` above makes sense.